### ISSUE TITLE
Paramiko 2.2.2 and 2.3.0 fail to import if gssapi is also installed.

### ISSUE THREAD

**Issue by JohnMalmberg** [paramiko/paramiko#1069]
Latest version that works is 2.2.1.

~~~
# pip freeze | grep gssapi
gssapi==1.2.0

# pip freeze | grep paramiko
paramiko==2.3.0

# python2
Python 2.7.5 (default, Nov  6 2016, 00:28:07)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import paramiko
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/paramiko/__init__.py", line 31, in <module>
    from paramiko.transport import SecurityOptions, Transport
  File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 38, in <module>
    from paramiko.auth_handler import AuthHandler
  File "/usr/lib/python2.7/site-packages/paramiko/auth_handler.py", line 47, in <module>
    from paramiko.ssh_gss import GSSAuth, GSS_EXCEPTIONS
  File "/usr/lib/python2.7/site-packages/paramiko/ssh_gss.py", line 54, in <module>
    GSS_EXCEPTIONS = (gssapi.GSSException,)
AttributeError: 'module' object has no attribute 'GSSException'
>>>exit()
~~~

**Comment by ploxiln** [comment]
dup of #1068 
(but I'm not sure why that one was closed)

**Comment by JohnMalmberg** [comment]
I see it was closed because installing the PyPi python-gssapi module restores paramiko loading.

This is still a bug because we do not install gssapi for paramiko, it is installed because ipa-client requires it.

Having the wrong gssapi installed instead of the expected one should still not stop paramiko from being imported.

**Comment by bitprophet** [comment]
> Having the wrong gssapi installed instead of the expected one should still not stop paramiko from being imported.

How do you figure that? If the two happen to have the same module name but differing APIs (as they appear to) they're effectively still different modules and lacking the correct one is the same as any other dependency error.

As before this is really a dupe, but of #861 - where we'll explicitly allow using either module and try to detect which one is in play. Hoping to get that out in 2.4.

**Comment by gianlucadardia** [comment]
I've solved removing python-gssapi module:
yum remove python-gssapi.x86_64

**Comment by JakeDEvans** [comment]
Not an option for us, yum remove python-gssapi is an ipa-client require module.

**Comment by athmane** [comment]
I've already addressed this issue in Fedora package with the following patch (should be temporary), the only downside is if you need GSSAPI.

https://src.fedoraproject.org/rpms/python-paramiko/blob/master/f/paramiko-2.3.1-disable-gssapi-on-unsupported-version.patch

**Comment by JakeDEvans** [comment]
thanks but your patch didn't work for me, centos 7.4, ipa-client 4.5, ansble 2.4.2

**Comment by WesleyBlancoYuan** [comment]
I removed two packages with `gss-api` in its name:

 - `cyrus-sasl-gssapi`
 - `python-gssapi`

I am with CentOS 7. Now all is fine. It seems that Python 2.7 is confused by the same name.

**Comment by b2sweety** [comment]
I am facing an issue " AttributeError: 'module' object has no attribute 'GSSException'" using ansible 2.4.3 when running the playbooks on any Network device (Cisco, Fortinet, Juniper). 

ansible 2.4.3.0
config file = /etc/ansible/ansible.cfg
configured module search path = [u'/user/ycaty/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
ansible python module location = /usr/lib/python2.7/site-packages/ansible
executable location = /usr/bin/ansible
python version = 2.7.5 (default, Nov 6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]

pip freeze | grep gssapi
gssapi==1.2.0
pip freeze | grep paramiko
paramiko==2.4.1
python --version
Python 2.7.5

@JakeDEvans Do we need  python2-ipaclient for ansible to work? I do not want to remove it and end up with having to install everything again. 
Can someone suggest a fix for the issue? All scripts are failing.

**Comment by JakeDEvans** [comment]
you must reinstall ansible in a virtual environment to work with freeipa clients.

**Comment by Numblesix** [comment]
Having the same issue again on fresh centos 7.5 install running ansible 2.6.5 and ipa server 4.5.4

**Comment by Deepakwani93** [comment]
pip uninstall gssapi woks for me
I was using paramiko version 2.4.2

**Comment by bitprophet** [comment]
Hopefully solved by #1311? which should get merged RSN.
